### DISCUSSION TITLE
how to clear TempData after loggoff?

### DISCUSSION THREAD

**Discussion by MRSheikh** [OrchardCMS/OrchardCore#9887] [Q&A]
hi.
I want to clear all TempData and other variables used in my module after the logoff post method of OrchardCore . 
I call logoff post like this:
```
<form class="" method="post" action="/Users/LogOff">
                      <button class="btn btn-light" type="submit">exit</button>
                      {% antiforgerytoken %}
 </form>

 how clear custom data after that?

**Comment by hishamco** [comment]
Use `TempData.Clear()`

**Reply by MRSheikh** [reply]
> TempData.Clear()

are you kidding? 
I know about `TempData.Clear()` but where I could call that method? the `LogOff` method is in the `Users` controller of OC source code.

**Reply by hishamco** [reply]
> I know about TempData.Clear()

You didn't mention that previously :) anyhow backing back to your comment we may need to add something similar to `ILoginFormEvent` but for logout, then you can register your implementation. @agriffard do we have something similar or shall I submit a PR for it?
